glog_test.go

```diff
@@ -531,3 +535,31 @@ func BenchmarkUseFsync(b *testing.B) {
 		Info("test")
 	}
 }
+
+func TestCleanOldLogs(t *testing.T) {
+	setFlags()
+	defer func() {
+		innerlogging.maxRetentionNum = 0
+	}()
+	innerlogging.maxRetentionNum = 2
+	f := flag.NewFlagSet("", flag.ExitOnError)
+	InitWithFlag(f)
+	f.Parse([]string{""})
+	StartWorker(time.Second)
+	innerlogging.file = [numSeverity]flushSyncWriter{}
+	innerlogging.createFiles(fatalLog)
+	for i := 0; i < 2; i++ {
+		for _, file := range innerlogging.file {
+			file.(*syncBuffer).rotateFile(time.Now())
+		}
+	}
+	//make sure there are only 2 file after 1s
+	time.Sleep(3 * time.Second)
```

Comment:
Pls add more test cases for different retention number

Reply:
will do